### DISCUSSION TITLE
Extract parameter names of a kernel

### DISCUSSION THREAD

**Discussion by chrissles** [GPflow/GPflow#1634] [Q&A]
Does anyone know how I can extract names of parameters of a kernel in gpflow, e.g. variance and lengthscale of gpflow.kernels.Matern32()?
With print_summary I get a column with these names but I cannot extract the name from the object. I would like to know which paramters a kernel has without looking into the doc, such that I can automatically set the best parameters after training a model multiple times.

**Comment by st--** [comment]
Hi @chrissles ! Does [how to copy (hyper)parameter values between models](https://gpflow.readthedocs.io/en/master/notebooks/intro_to_gpflow2.html#Copying-(hyper)parameter-values-between-models) answer your question?
